src/filesystem.cpp

```diff
@@ -1064,43 +1064,31 @@ std::string get_exe_dir()
 	std::string exe = filesystem::get_program_invocation("wesnoth-"+version);
 	bfs::path search = bp::search_path(exe).string();
 	if(!search.string().empty()) {
-		return search.parent_path().string();
+		return {search.parent_path().string(), exe};
 	}
 
 	// versionless
 	exe = filesystem::get_program_invocation("wesnoth");
 	search = bp::search_path(exe).string();
 	if(!search.string().empty()) {
-		return search.parent_path().string();
+		return {search.parent_path().string(), exe};
 	}
 
 	// return the current working directory
-	return get_cwd();
+	return {get_cwd(), "wesnoth"};
 #endif
 }
 
 std::string get_wesnothd_name()
 {
-	std::string exe_dir = get_exe_dir()+"/";
-	std::string version = get_version_path_suffix();
-
-	// many potential variations used by linux packagers
-	std::vector<std::string> wesnothds = {
-		exe_dir+get_program_invocation("wesnothd"+version),
-		exe_dir+get_program_invocation("wesnothd-"+version),
-		exe_dir+get_program_invocation("wesnothd_"+version),
-		exe_dir+get_program_invocation("wesnothd_devel"),
-		exe_dir+get_program_invocation("wesnothd-devel"),
-		exe_dir+get_program_invocation("wesnothd")
-	};
-
-	for(const std::string& wesnothd : wesnothds) {
-		if(file_exists(wesnothd)) {
-			return wesnothd;
-		}
-	}
-
-	return "";
+	auto [exe_dir, exe] = get_exe_dir();
+	// macOS doesn't call the wesnoth client executable "wesnoth"
+	// otherwise, add any suffix after the "wesnoth" part of the executable name to wesnothd's name
+#ifdef __APPLE__
+	return exe_dir + "/" + get_program_invocation("wesnothd");
```

Comment:
Is this right? What does `get_program_invocation` do?

Reply:
If built in Debug mode, append `-debug`. If Windows, also append `.exe`.

Reply:
I don't think Xcode appends `-debug`.

Reply:
This PR changes it to only append `-debug` if it's Windows, but I'm not sure if that's needed at all. cmake never did that, scons used to but doesn't anymore, and Iris confirmed that xcode/macOS doesn't either. So it's just a question of whether the cmake-generated Visual Studio project files still do it or not.

Reply:
Would be better if this used an Xcode specific define.

Alternatively no special logic and check if the file exists and have a fixed "wesnothd" fallback.